### DISCUSSION TITLE
CSR in NextJs

### DISCUSSION THREAD

**Discussion by Maanu07** [vercel/next.js#45703] [Help]
Suppose in my homepage i have `<Link/>`  component to my `about` page. So if I visit the about `page` using the `<Link/>` component then in that case do `nextjs` perform `Client side rendering` to display my `About` page? Please let me know.

**Comment by icyJoseph** [comment]
Navigation using the `next/link` components is an SPA transition. The page doesn't unload, rather, the framework "swap" the components for the current page, with the components for the "next" page. 

When the `next/link` mounts, next.js fetches the JS and JSON (if any) that drives that page, so that when you click on it, the client side rendering kicks in, the JSON is passed as props to the page component and the rendering happens.

For direct navigation, if the page was pre-rendered, or if it uses GetServerSideProps, the HTML is sent from the server to the client, but still the client does something called hydration, where interactivity is added to the page.
